### PR TITLE
Ignore empty keys in QueryStringValueProvider

### LINKED ISSUE #8484: Querystring with empty key will fill every parameter with the specified value
#### Steps to reproduce (preferably a link to a GitHub repo with a repro project):
using the query with a empty key ("?=bug") will result in all* parameters being filled with the value ("bug") specified
[https://github.com/KLuuKer/bug-url-query-param](https://github.com/KLuuKer/bug-url-query-param)

#### Description of the problem:
when sending the following url **?=bug** every parameter will get filled with the string "bug" (except those with the FromQuery attribute that has the Name= property specified)

bug example

#### Version of `Microsoft.AspNetCore.App`: 
```
<Project Sdk="Microsoft.NET.Sdk.Web">
<TargetFramework>netcoreapp2.1</TargetFramework>
<PackageReference Include="Microsoft.AspNetCore.App" />
```
happening on production box with 'hosting runtime bundle v2.1.4'
and when running debug with visual studio 15.9.0 preview 2

### PR DESCRIPTION
**Pull request by pranavkm**
Fixes #8484

### CODE REVIEW

**Comment by pranavkm** [comment]
Note that this is going in to master (3.0)

File: src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/QueryStringValueProvider.cs [hunk 1]
```diff
@@ -86,6 +86,11 @@ public override ValueProviderResult GetValue(string key)
                 throw new ArgumentNullException(nameof(key));
             }
 
+            if (key.Length == 0)
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/QueryStringValueProvider.cs]: This needs an explanation as to why in the comments 😆 

Should we also consider changes to the other value providers?

Comment (pranavkm) [reply src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/QueryStringValueProvider.cs]: Considering this is going in 3.0, I find it more palatable to make it for all our value providers. Makes the behavior more predictable.

Comment (rynowak) [reply src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/QueryStringValueProvider.cs]: Yeah I guess I was wondering if it's possible for our other value providers. Like, we should find out if `Form` is hardened against empty keys, n, stuff like that.

Comment (rynowak) [reply src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/QueryStringValueProvider.cs]: Is there an answer to this cool question?

Comment (pranavkm) [reply src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/QueryStringValueProvider.cs]: This change hardens the following value providers: `FormValueProvider`, `QueryStringValueProvider`, `RouteValueProvider`. I left the jquery providers as is. I didn't want to spend time trying to see what the behavior is across versions.

**Pull request merged**